packages/rum/src/domain/rumEventsCollection/resource/resourceCollection.ts

```diff
@@ -17,102 +17,107 @@ import {
   computePerformanceResourceDuration,
   computeResourceKind,
   computeSize,
+  isRequestKind,
 } from '../resourceUtils'
 import { matchRequestTiming } from './matchRequestTiming'
 
 export function startResourceCollection(lifeCycle: LifeCycle, configuration: Configuration, session: RumSession) {
-  const handler = (startTime: number, rawRumEvent: RawRumEvent) =>
-    lifeCycle.notify(LifeCycleEventType.RAW_RUM_EVENT_COLLECTED, {
-      rawRumEvent,
-      startTime,
-    })
   lifeCycle.subscribe(LifeCycleEventType.REQUEST_COMPLETED, (request: RequestCompleteEvent) => {
-    handleRequest(lifeCycle, session, handler, request)
+    if (session.isTrackedWithResource()) {
+      lifeCycle.notify(LifeCycleEventType.RAW_RUM_EVENT_COLLECTED, processRequest(request))
+      lifeCycle.notify(LifeCycleEventType.RESOURCE_ADDED_TO_BATCH)
```

Comment:
NIT for another PR: `RESOURCE_ADDED_TO_BATCH` could probably be replaced with a `RAW_RUM_EVENT_COLLECTED` usage + test on evt.category

Reply:
yep, I've listed some cleanup [here](https://datadoghq.atlassian.net/browse/RUMF-748)